### ISSUE TITLE
Allow Cross client credentials

### ISSUE THREAD

**Issue by mohsinhijazee** [mintel/dex-k8s-authenticator#140]
Currently, when following is added into scopes:
```yaml
    scopes: 
      - openid
      - profile
      - email
      - offline_access
      - groups
      - audience:server:client_id:kubernetes
```
This gets ignored altogether and the genrated auth URL conly contains following as scopes:

```
&scope=openid+profile+email+offline_access+groups
```

Whereas the `example-app` on which this authenticator is based on, perfectly supports this behavior. Further documentation on dex:

* [List of supported scopes](https://dexidp.io/docs/custom-scopes-claims-clients/#scopes)
* [Supports for cross client credentials](https://dexidp.io/docs/custom-scopes-claims-clients/#cross-client-trust-and-authorized-party)

### Why?
As the k8s API server only allows one client application to be used as authenticator (via the `--oidc-*` set of flags), we don't want to expose the main authenticator application itself everywhere and rather create another client application with `client_id kubectl` for example and then allow it to obtain tokens on behalf of the main application. In case of a breach, only this application would have to be revoked/adjusted and everything else would keep working fine.

**Comment by nabadger** [comment]
Will look into this :+1:

**Comment by betermieux** [comment]
Pull Request #129 (which supports custom scopes) is merged into master but is not yet released (with latest 1.2.0). So, releasing a new version should close this issue.
I am waiting for it, too. ;-)

**Comment by nabadger** [comment]
@betermieux I've released new versions today (`1.4.0` being the latest).

Sorry for delay!
